### PR TITLE
Add support for GlassFish

### PR DESCRIPTION
**Pull request by iNikem**

### CODE REVIEW

**Review by vovencij — CHANGES_REQUESTED** [review]

File: matrix/src/payara-dockerfile.template [hunk 1]
```diff
@@ -1,3 +1,5 @@
 FROM payara/server-full:@version@
 
+RUN rm ${PAYARA_DIR}/glassfish/modules/phonehome-bootstrap.jar
```
[the remark this answers is not in the document]
Comment (iNikem) [reply matrix/src/payara-dockerfile.template:3]: Yes, these are two different "phone home". Payara does its own and Hazelcast does its own.

File: matrix/src/main/webapp/WEB-INF/glassfish-web.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<glassfish-web-app>
+  <context-root>/</context-root>
```
[the remark this answers is not in the document]
Comment (iNikem) [reply matrix/src/main/webapp/WEB-INF/glassfish-web.xml:2]: All test apps are now deployed under non-root context

File: smoke-tests/src/test/java/com/splunk/opentelemetry/SmokeTest.java [hunk 3]
```diff
@@ -107,9 +110,17 @@ void startTarget(String targetImageName) {
             .withEnv("OTEL_BSP_SCHEDULE_DELAY", "10")
             .withEnv("OTEL_EXPORTER_JAEGER_ENDPOINT", "http://collector:14268/api/traces")
             .withEnv(getExtraEnv());
+    WaitStrategy waitStrategy = getWaitStrategy();
+    if (waitStrategy != null) {
+      target = target.waitingFor(waitStrategy);
+    }
```
[the remark this answers is not in the document]
Comment (iNikem) [reply smoke-tests/src/test/java/com/splunk/opentelemetry/SmokeTest.java:116]: I am not convinced that this alternative is more readable...

**Review by vovencij — APPROVED** [review]
Will look into it next.

File: matrix/src/main/webapp/WEB-INF/glassfish-web.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (vovencij) [reply matrix/src/main/webapp/WEB-INF/glassfish-web.xml:2]: Shouldn't you also change this file?

File: smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+  @Override
+  protected Map<String, String> getExtraEnv() {
+    return Map.of("HZ_PHONE_HOME_ENABLED", "false");
+  }
+
+  @Override
+  protected WaitStrategy getWaitStrategy() {
+    return Wait.forLogMessage(".*app was successfully deployed.*", 1)
+        .withStartupTimeout(Duration.ofMinutes(15));
```
Comment (mateuszrzeszutek) [inline smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java:57]: That's a huge timeout. Does it really take this long to start?

**Review by mateuszrzeszutek — APPROVED** [review]

File: smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java (hunk 4 above)
Comment (iNikem) [reply smoke-tests/src/test/java/com/splunk/opentelemetry/GlassFishSmokeTest.java:57]: No, it does not. Left from local testing, will decrease separately

**Pull request merged**